src/managers/world-manager.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+var WorldApi = require('../service/world-api-adapter');
+var RunManager =  require('./run-manager');
+var AuthManager = require('./auth-manager');
+// var _pick = require('../util/object-util')._pick;
+var worldApi;
+
+// var defaults = {
+//  account: '',
+//  project: '',
+//  group: '',
+//  transport: {
+//  }
+// };
+
+
+function buildStrategy(worldId, dtd) {
+
+    return function Ctor(runService, options) {
+        this.runService = runService;
+        this.options = options;
+
+        $.extend(this, {
+            reset: function () {
+                throw new Error('not implementd. Need api changes');
+            },
+
+            getRun: function () {
+                var _this = this;
+                return worldApi.getCurrentRunId({ filter: worldId })
+                    .then(function (runId) {
+                        return _this.runService.load(runId);
+                    })
+                    .then(function (run) {
+                        dtd.resolve.call(this, run, _this.runService);
```

Comment:
According to http://api.jquery.com/deferred.then/ the second parameter to 'then' is always a failure handler - this contradict the docs for `resolve` but matches the general javascript promise spec I think.